### PR TITLE
Homework for Lesson 4

### PR DESCRIPTION
**Pull request by filkodaria**
Homework_4: task_1, task_2, task_3.
P.S. I've tried to cover all possible cases with invalid values in task_3 (file hw_4.3.js).

### CODE REVIEW

File: hw_4/hw_4.2.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		console.log('Keep calm and look Culture channel.');
+		break;
+	}
+	default: {
+		console.log('Technical work...1');
+	}
+}
+
+// subtask 2
+if (typeof age != 'number' || isNaN(age)) {
```
Comment (anatoly-karpovich) [inline hw_4/hw_4.2.js]: typeof age is enough here, no need to have isNaN for 2nd subtask

File: hw_4/hw_4.2.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+} else if (age < age_2) {
+	console.log(`You don\'t have access cause your age is ${age}. It\'s less then ${age_2}.`);
+} else if (age >= age_2 && age < age_3) {
+	console.log('Welcome!');
+} else if (age >= age_3) {
+	console.log('Keep calm and look Culture channel.');
+}
+
+// subtask 3
+if (Boolean(+age) === false) {
```
Comment (anatoly-karpovich) [inline hw_4/hw_4.2.js]: 1. Technical work...3 for age === 0, cause 0 is falsy value. Need to use isNaN here
2. Boolean(+age) === false can be changed to !(+age)

File: hw_4/hw_4.3.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+*/
+
+let age;
+age = prompt('Enter your age.');
+
+const age_2 = 18;
+const age_3 = 60;
+
+if (!Number.isInteger(+age) || age < 0 || Boolean(+age) === false) {
```
Comment (anatoly-karpovich) [inline hw_4/hw_4.3.js]: Same situation as for the above comment. Please update checks

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

File: hw_4/hw_4.2.js (hunk 1 above)
Comment (filkodaria) [reply hw_4/hw_4.2.js]: Ок. But NaN has also type number and we need to handle this case (age = NaN). Why not?

File: hw_4/hw_4.3.js (hunk 3 above)
Comment (filkodaria) [reply hw_4/hw_4.3.js]: Not at all.
I covered cases for age = ‘’, age = ‘  ‘, age = null, etc.
It works quite well. 
Could you launch it via Live Server?

File: hw_4/hw_4.2.js (hunk 2 above)
Comment (filkodaria) [reply hw_4/hw_4.2.js]: Thanks! I’ll think over it.

File: hw_4/hw_4.2.js (hunk 1 above)
Comment (anatoly-karpovich) [reply hw_4/hw_4.2.js]: Cause as per requirements need to check only for number type

File: hw_4/hw_4.3.js (hunk 3 above)
Comment (anatoly-karpovich) [reply hw_4/hw_4.3.js]: As per my understanding you need the next flow:
1. Check isNaN(age). If true - provide error
2. Convert age to number (if it is number as well - nothing will happen)
3. Do regular checks for different ages

File: hw_4/hw_4.2.js (hunk 1 above)
Comment (filkodaria) [reply hw_4/hw_4.2.js]: So, in such way we don't have any output for case age = NaN, it is skipped => is it OK?
@anatoly-karpovich

Comment (anatoly-karpovich) [reply hw_4/hw_4.2.js]: As per the task - yes

Преобразовать написанный код в task 1 так, чтобы сначала проверялся тип данных. И если он не number - кидалась ошибка в консоль. Проверить работу кода на следующих данных 17, 18, 61, "2", "aaa"

Comment (filkodaria) [reply hw_4/hw_4.2.js]: done)

File: hw_4/hw_4.2.js (hunk 2 above)
Comment (filkodaria) [reply hw_4/hw_4.2.js]: done

**Review by VladimirLQA — APPROVED** [review]

**Review by anatoly-karpovich — APPROVED** [review]

File: hw_4/hw_4.3.js (hunk 3 above)
Comment (filkodaria) [reply hw_4/hw_4.3.js]: done

**Comment by filkodaria** [comment]
@anatoly-karpovich 
I've added changes to the files hw_4.2.js and hw_4.3.js
Please, check)

**Review by anatoly-karpovich — APPROVED** [review]

**Pull request merged**